mempool/src/pool/feerate_points.rs

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) 2021-2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use std::{
+    cmp::{Eq, Ord},
+    collections::BTreeMap,
+    ops::{Add, Div, Mul, Sub},
+};
+
+use common::primitives::Amount;
+
+use super::fee::Fee;
+
+pub fn linear_interpolation<X, Y>(p0: (X, Y), p1: (X, Y), x: X) -> Option<Y>
+where
+    X: Add<Output = X> + Sub<Output = X> + Eq + Ord + Copy,
+    Y: Mul<Output = Y> + Add<Output = Y> + Div<Output = Y> + Copy + From<X>,
```

Comment:
Note that all these bounds signify unchecked arithmetic. If the function is supposed to check for overflows of these operations, I'd expect to see something like `X: Op<Output = Option<X>>`